library/core/class.format.php

```diff
@@ -955,6 +958,35 @@ protected static function isHtml($Text) {
     }
 
     /**
+     * Checks to see if $html has a spoiler element and replaces the element.
+     *
+     * @param string $html A html-formatted string
+     * @return string
+     */
+    protected static function replaceSpoilers($html) {
+        // Check if $html contains spoiler classes before we do the heavy lifting.
+        $found = false;
+        foreach(Gdn_Format::$spoilerClasses as $spoiler) {
```

Comment:
Not a fan of this loop. I'd maybe just hardcode the spoiler regex for speed: `#class="(User)?Spoiler"#i`